gempak/ush/gfs_meta_comp.sh

```diff
@@ -1,184 +1,153 @@
 # GFS model run.
 #
-# Log :
-# J. Carr/HPC    5/12/97   Developed Script
-# J. Carr/HPC    8/05/98   Changed map to medium resolution and redefined yesterday code
-# J. Carr/HPC    2/01/99   Changed skip to 0
-# J. Carr/HPC    4/12/99   Added gfs model out to 84 hours.
-# J. Carr/HPC       6/99   put a filter on map
-# J. Carr/HPC     4/2000   Upped the eta comp to 60 hrs.
-# J. Carr/HPC     2/2001   Edited to run on the IBM.
-# J. Carr/HPC     5/2001   Added a mn variable for a/b side dbnet root variable.
-# J. Carr/HPC     7/2001   Added more comparison times.
-# J. Carr/HPC     7/2001   Converted to a korn shell prior to delivering script to Production.
-# J. Carr/HPC     7/2001   Submitted.
-# J. Carr/HPC    11/2004   Changed all eta/ETA entries to nam/NAM.
-#                          Inserted a ? in all title/TITLE lines.
-#
 # Set up Local Variables
 #
-set -x
-#
-export PS4='COMP:$SECONDS + '
-rm -Rf $DATA/COMP $DATA/GEMPAK_META_COMP
-mkdir -p -m 775 $DATA/COMP  $DATA/GEMPAK_META_COMP
-cd $DATA/COMP
-cp ${HOMEgfs}/gempak/fix/datatype.tbl datatype.tbl
 
-export COMPONENT=${COMPONENT:-atmos}
+source "${HOMEgfs}/ush/preamble.sh"
+
+rm -Rf "${DATA}/COMP" "${DATA}/GEMPAK_META_COMP"
+mkdir -p -m 775 "${DATA}/COMP"  "${DATA}/GEMPAK_META_COMP"
+cd "${DATA}/COMP" || exit 2
+cp "${HOMEgfs}/gempak/fix/datatype.tbl" datatype.tbl
 
 mdl=gfs
 MDL=GFS
 metatype="comp"
 metaname="${mdl}_${metatype}_${cyc}.meta"
 device="nc | ${metaname}"
-PDY2=$(echo $PDY | cut -c3-)
-#
-#XXW export MODEL=$COMROOT/nawips/prod
-# BV export MODEL=$COMROOT/nawips/${envir}
-# BV export HPCGFS=${MODEL}/${mdl}.$PDY
-export HPCGFS=${COMINgempak}/${mdl}.${PDY}/${cyc}/${COMPONENT}/gempak
-export COMIN00=${COMINgempak}/${mdl}.${PDY}/00/${COMPONENT}/gempak
-export COMIN06=${COMINgempak}/${mdl}.${PDY}/06/${COMPONENT}/gempak
-export COMIN12=${COMINgempak}/${mdl}.${PDY}/12/${COMPONENT}/gempak
-export COMIN18=${COMINgempak}/${mdl}.${PDY}/18/${COMPONENT}/gempak
-if [ ${cyc} -eq 00 ] ; then
-   cp $COMIN00/gfs_${PDY}00f* $DATA/GEMPAK_META_COMP
-elif [ ${cyc} -eq 06 ] ; then
-   cp $COMIN00/gfs_${PDY}00f* $DATA/GEMPAK_META_COMP
-   cp $COMIN06/gfs_${PDY}06f* $DATA/GEMPAK_META_COMP
-elif [ ${cyc} -eq 12 ] ; then
-   cp $COMIN00/gfs_${PDY}00f* $DATA/GEMPAK_META_COMP
-   cp $COMIN06/gfs_${PDY}06f* $DATA/GEMPAK_META_COMP
-   cp $COMIN12/gfs_${PDY}12f* $DATA/GEMPAK_META_COMP
-elif [ ${cyc} -eq 18 ] ; then
-   cp $COMIN00/gfs_${PDY}00f* $DATA/GEMPAK_META_COMP
-   cp $COMIN06/gfs_${PDY}06f* $DATA/GEMPAK_META_COMP
-   cp $COMIN12/gfs_${PDY}12f* $DATA/GEMPAK_META_COMP
-   cp $COMIN18/gfs_${PDY}18f* $DATA/GEMPAK_META_COMP
-fi
-export COMIN=$DATA/GEMPAK_META_COMP
 
-#XXW export HPCNAM=${MODEL}/nam.$PDY
-#XXW export HPCNGM=${MODEL}/ngm.$PDY
-# BV export HPCNAM=$COMROOT/nawips/prod/nam.$PDY
-export HPCNAM=${COMINnam}.$PDY/gempak
+export COMIN="gfs.multi"
+mkdir "${COMIN}"
+for cycle in $(seq -f "%02g" -s ' ' 0  "${STEP_GFS}" "${cyc}"); do
```

Comment:
I don't see `${STEP_GFS}` defined prior to this line. Is it defined in preamble.sh?

Reply:
`STEP_GFS` is defined in `config.base`